File: ios/SpotifyAlarm/Spotify/SpotifyConfig.swift
import Foundation

/// Public Spotify app credentials. The client ID is safe to ship in the app
/// when using PKCE — no client secret is ever embedded.
///
/// Configure in Spotify Developer Dashboard:
///   • Redirect URI: spotifyalarm://callback
///   • iOS Bundle ID: com.daviduek.spotifyalarm
enum SpotifyConfig {
    /// Read from Info.plist key `SpotifyClientID`, falling back to the
    /// compile-time default. Override per-build with an .xcconfig.
    static var clientID: String {
        if let value = Bundle.main.object(forInfoDictionaryKey: "SpotifyClientID") as? String,
           !value.isEmpty {
            return value
        }
        return "REPLACE_WITH_YOUR_CLIENT_ID"
    }

    static let redirectURI = URL(string: "spotifyalarm://callback")!
    static let redirectScheme = "spotifyalarm"

    static let tokenSwapURL: URL? = nil  // optional backend swap; nil = direct PKCE
    static let tokenRefreshURL: URL? = nil

    static let scopes: [String] = [
        "user-read-private",
        "user-read-email",
        "user-read-playback-state",
        "user-modify-playback-state",
        "user-read-currently-playing",
        "streaming",
        "playlist-read-private",
        "playlist-read-collaborative",
        "user-library-read",
        "user-read-recently-played",
        "app-remote-control"
    ]

    static var scopeString: String { scopes.joined(separator: " ") }

    static let accountsBaseURL = URL(string: "https://accounts.spotify.com")!
    static let apiBaseURL = URL(string: "https://api.spotify.com/v1")!
}

File: apps/mobile/src/services/spotify/spotifyConfig.ts
import { makeRedirectUri } from 'expo-auth-session';

import { APP_SCHEME, spotifyClientId } from '../config';

/** OAuth 2.0 Authorization Code + PKCE (spec §10). No client secret exists anywhere in the app. */
export const SPOTIFY_DISCOVERY = {
  authorizationEndpoint: 'https://accounts.spotify.com/authorize',
  tokenEndpoint: 'https://accounts.spotify.com/api/token',
};

export const SPOTIFY_SCOPES = [
  'user-read-private',
  'user-read-playback-state',
  'user-modify-playback-state',
  'playlist-read-private',
  'playlist-read-collaborative',
  'user-library-read',
];

/** Must match the Redirect URI registered in the Spotify Developer Dashboard: wake://spotify-callback */
export const SPOTIFY_REDIRECT_URI = makeRedirectUri({ scheme: APP_SCHEME, path: 'spotify-callback' });

export const SPOTIFY_API_BASE = 'https://api.spotify.com/v1';

export const isSpotifyConfigured = (): boolean => spotifyClientId.length > 0;
export { spotifyClientId };
